app/webpacker/components/wca/FormBuilder/provider/FormObjectProvider.jsx

```diff
@@ -0,0 +1,110 @@
+import React, {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useReducer,
+} from 'react';
+import _ from 'lodash';
+import { changesSaved, setErrors } from '../store/actions';
+import formReducer from '../store/reducer';
+
+const FormContext = createContext();
+
+export default function FormObjectProvider({
+  children,
+  initialObject,
+}) {
+  const initialState = useMemo(() => ({
+    object: initialObject,
+    initialObject,
+    errors: null,
+  }), [initialObject]);
+
+  const [formState, dispatch] = useReducer(formReducer, initialState);
```

Comment:
Might as well define
```
const createState = (initialObject) => ({
  object: initialObject,
  initialObject,
  errors: null,
})
```
_outside_ the component, then do
```
const [formState, dispatch] = useReducer(formReducer, initialState, createState);
```

Reply:
Cool! I didn't know this was a thing, thanks for pointing it out :)